### PR TITLE
Declare Variant "adjust_args" test

### PR DESCRIPTION
**Pull request by nolanbaker31**
Test passes with most recent LLVM on summit. LMK thoughts.

### CODE REVIEW

**Comment by nolanbaker31** [comment]
Test now fails on LLVM w/ 30-40 errors (different each run)

**Comment by andrewkallai** [comment]
Note: functionality of dispatch construct is required for this test.
Frontier
clang version 17.0.0 (https://github.com/llvm/llvm-project.git 8b4a27f410ba42b2e19651ddafc60cd878d0963c)
CRASH
https://clang.llvm.org/docs/OpenMPSupport.html#openmp-5-1-implementation-details
https://github.com/llvm/llvm-project/issues/62389
Indication that LLVM developers have not fully implemented "dispatch construct and function variant argument adjustment."
Though it seems declare variant is implemented.
https://github.com/llvm/llvm-project/issues/63871
Potential bug may prevent testing any existing implementation.

File: tests/5.1/declare_variant/test_declare_variant_adjust_args.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int arr[N]; // implicit map array
```
Comment (spophale) [inline tests/5.1/declare_variant/test_declare_variant_adjust_args.c:24]: malloc arr

File: tests/5.1/declare_variant/test_declare_variant_adjust_args.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+int test_wrapper() { 
+    errors = 0;
+    add(arr);
+    for(i = 0; i < N; i++){
+        OMPVV_TEST_AND_SET_VERBOSE(errors, arr[i] != i+1);
+    } 
+    OMPVV_ERROR_IF(errors > 0, "Base function is not working properly");
+
```
Comment (spophale) [inline tests/5.1/declare_variant/test_declare_variant_adjust_args.c:51]: #pragma omp target enter data(to:arr[0:N])

File: tests/5.1/declare_variant/test_declare_variant_adjust_args.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        will be converted in the same manner that a use_device_ptr clause on a target data construct converts 
+        its pointer list items into device pointers." */
+    
+    #pragma omp target parallel for map(tofrom: errors)
+    for(i = 0; i < N; i++){
+        if(arr[i] != i+3){
+            errors++;
+        }
+    }
+    OMPVV_ERROR_IF(errors > 0, "Dispatch w/ depend is not working properly");
```
Comment (spophale) [inline tests/5.1/declare_variant/test_declare_variant_adjust_args.c:64]: Before OMPVV_ERROR_ ...
#pragma omp target exit data(delete:arr[0:N])

File: tests/5.1/declare_variant/test_declare_variant_adjust_args.c [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+#pragma omp declare variant(add_two) match(construct={dispatch}) adjust_args(need_device_ptr:arr)
+void add(int *arr){
+    for (int i = 0; i < N; i++){ // Base function adds 1 to array values
+        arr[i] = i+1;
+    }
+}
+
+void add_two(int *arr){
+    #pragma omp target parallel for
```
Comment (spophale) [inline tests/5.1/declare_variant/test_declare_variant_adjust_args.c:38]: Here check if arr is NULL.

**Review by spophale — APPROVED** [review]

**Pull request merged**